src/geanyobject.c

```diff
@@ -131,231 +78,154 @@ static gboolean boolean_handled_accumulator(GSignalInvocationHint *ihint, GValue
-	register GCClosure *cc = (GCClosure*) closure;
-	register gpointer data1, data2;
-	gboolean v_return;
-
-	g_return_if_fail(return_value != NULL);
-	g_return_if_fail(n_param_values == 3);
-
-	if (G_CCLOSURE_SWAP_DATA(closure))
-	{
-		data1 = closure->data;
-		data2 = g_value_peek_pointer(param_values + 0);
-	}
-	else
-	{
-		data1 = g_value_peek_pointer(param_values + 0);
-		data2 = closure->data;
-	}
-	callback = (GeanyMarshalFunc_BOOLEAN__POINTER_POINTER)
-		(marshal_data ? marshal_data : cc->callback);
-
-	v_return = callback(data1,
-					   g_value_get_pointer(param_values + 1),
-					   g_value_get_pointer(param_values + 2),
-					   data2);
-
-	g_value_set_boolean(return_value, v_return);
-}
-
-
 static void create_signals(GObjectClass *g_object_class)
 {
 	/* Document signals */
 	geany_object_signals[GCB_DOCUMENT_NEW] = g_signal_new (
 		"document-new",
 		G_OBJECT_CLASS_TYPE (g_object_class),
 		G_SIGNAL_RUN_FIRST,
-		G_STRUCT_OFFSET (GeanyObjectClass, document_new),
-		NULL, NULL,
-		g_cclosure_marshal_VOID__POINTER,
+		0, NULL, NULL, g_cclosure_marshal_VOID__BOXED,
 		G_TYPE_NONE, 1,
-		G_TYPE_POINTER);
+		GEANY_TYPE_DOCUMENT);
 	geany_object_signals[GCB_DOCUMENT_OPEN] = g_signal_new (
 		"document-open",
 		G_OBJECT_CLASS_TYPE (g_object_class),
 		G_SIGNAL_RUN_FIRST,
-		G_STRUCT_OFFSET (GeanyObjectClass, document_open),
-		NULL, NULL,
-		g_cclosure_marshal_VOID__POINTER,
+		0, NULL, NULL, g_cclosure_marshal_VOID__BOXED,
 		G_TYPE_NONE, 1,
-		G_TYPE_POINTER);
+		GEANY_TYPE_DOCUMENT);
 	geany_object_signals[GCB_DOCUMENT_RELOAD] = g_signal_new (
 		"document-reload",
 		G_OBJECT_CLASS_TYPE (g_object_class),
 		G_SIGNAL_RUN_FIRST,
-		G_STRUCT_OFFSET (GeanyObjectClass, document_reload),
-		NULL, NULL,
-		g_cclosure_marshal_VOID__POINTER,
+		0, NULL, NULL, g_cclosure_marshal_VOID__BOXED,
 		G_TYPE_NONE, 1,
-		G_TYPE_POINTER);
+		GEANY_TYPE_DOCUMENT);
 	geany_object_signals[GCB_DOCUMENT_BEFORE_SAVE] = g_signal_new (
 		"document-before-save",
 		G_OBJECT_CLASS_TYPE (g_object_class),
 		G_SIGNAL_RUN_FIRST,
-		G_STRUCT_OFFSET (GeanyObjectClass, document_before_save),
-		NULL, NULL,
-		g_cclosure_marshal_VOID__POINTER,
+		0, NULL, NULL, g_cclosure_marshal_VOID__BOXED,
 		G_TYPE_NONE, 1,
-		G_TYPE_POINTER);
+		GEANY_TYPE_DOCUMENT);
 	geany_object_signals[GCB_DOCUMENT_SAVE] = g_signal_new (
 		"document-save",
 		G_OBJECT_CLASS_TYPE (g_object_class),
 		G_SIGNAL_RUN_FIRST,
-		G_STRUCT_OFFSET (GeanyObjectClass, document_save),
-		NULL, NULL,
-		g_cclosure_marshal_VOID__POINTER,
+		0, NULL, NULL, g_cclosure_marshal_VOID__BOXED,
 		G_TYPE_NONE, 1,
-		G_TYPE_POINTER);
+		GEANY_TYPE_DOCUMENT);
 	geany_object_signals[GCB_DOCUMENT_FILETYPE_SET] = g_signal_new (
 		"document-filetype-set",
 		G_OBJECT_CLASS_TYPE (g_object_class),
 		G_SIGNAL_RUN_FIRST,
-		G_STRUCT_OFFSET (GeanyObjectClass, document_filetype_set),
-		NULL, NULL,
-		geany_cclosure_marshal_VOID__POINTER_POINTER,
+		0, NULL, NULL, g_cclosure_marshal_generic,
```

Comment:
shouldn't you simply pass `NULL` as the marshaller instead of explicitly passing the generic one?  Clearer for me (as I'm not fooled to see there's a specific one selected), shorter, and there doesn't seem to be no use for passing it explicitly, is it?